examples/HierarchicalOptimization_examples/BachmannJakStat/logLikelihood_BachmannJakStat.m

```diff
@@ -0,0 +1,182 @@
+            switch options.llh.distribution
+                case 'normal'
+                    logL = logL - 0.5*(sum(sum(nansum(bsxfun(@times,~isnan(D(cond).my),...
+                        log(2*pi*sigma2))+bsxfun(@rdivide,bsxfun(@power,y_ch,2),sigma2),1),3),2));
+                    if nargout > 1
+                        if cond == 3 && linRNAflag
+                            iy = [12:17];
+                            dlogL = dlogL - 0.5*(permute(sum(sum(nansum(...
+                                bsxfun(@times,permute(bsxfun(@rdivide,(1-bsxfun(@rdivide,...
+                                bsxfun(@power,y_ch(:,iy,:),2),sigma2(:,iy,:))),sigma2(:,iy,:)),[1,2,4,3]),dsigma2(:,iy,:,:)) -...
+                                bsxfun(@times,permute(2*bsxfun(@rdivide,y_ch(:,iy,:),sigma2(:,iy,:)),[1,2,4,3]),sol(cond).sy(:,iy,:))...
+                                ,1),4),2),[3,2,1]));
+                        elseif cond == 2
+                            iy = [1:10,12:20];
+                            dlogL = dlogL - 0.5*(permute(sum(sum(nansum(...
+                                bsxfun(@times,permute(bsxfun(@rdivide,(1-bsxfun(@rdivide,...
+                                bsxfun(@power,y_ch(:,iy,:),2),sigma2(:,iy,:))),sigma2(:,iy,:)),[1,2,4,3]),dsigma2(:,iy,:,:)) -...
+                                bsxfun(@times,1/log(10)*permute(2*bsxfun(@rdivide,y_ch(:,iy,:),sigma2(:,iy,:)),[1,2,4,3]),...
+                                bsxfun(@rdivide,sol(cond).sy(:,iy,:),sol(cond).y(:,iy)))...
+                                ,1),4),2),[3,2,1]));
+                            iy = 11;
+                            dlogL = dlogL - 0.5*(permute(sum(sum(nansum(...
+                                bsxfun(@times,permute(bsxfun(@rdivide,(1-bsxfun(@rdivide,...
+                                bsxfun(@power,y_ch(:,iy,:),2),sigma2(:,iy,:))),sigma2(:,iy,:)),[1,2,4,3]),dsigma2(:,iy,:,:)) -...
+                                bsxfun(@times,permute(2*bsxfun(@rdivide,y_ch(:,iy,:),sigma2(:,iy,:)),[1,2,4,3]),sol(cond).sy(:,iy,:))...
+                                ,1),4),2),[3,2,1]));
+                        else
+                            dlogL = dlogL - 0.5*(permute(sum(sum(nansum(...
+                                bsxfun(@times,permute(bsxfun(@rdivide,(1-bsxfun(@rdivide,...
+                                bsxfun(@power,y_ch,2),sigma2)),sigma2),[1,2,4,3]),dsigma2) -...
+                                bsxfun(@times,1/log(10)*permute(2*bsxfun(@rdivide,y_ch,sigma2),[1,2,4,3]),...
+                                bsxfun(@rdivide,sol(cond).sy,sol(cond).y))...
+                                ,1),4),2),[3,2,1]));
+                        end
+                    end
+                case 'laplace'
+                    logL = logL - sum(sum(nansum(bsxfun(@times,~isnan(D(cond).my),log(2*sigma2))+...
+                        bsxfun(@rdivide,abs(y_ch),sigma2),1),3),2);
+                    
+                    if nargout > 1
+                        if cond == 3 && linRNAflag
```

Comment:
same here. bonus points if you can directly encode the differences between normal and laplace. something like `bsxfun(@power,y_ch(:,iy,:),2),sigma2(:,iy,:)))` for normal and `bsxfun(@rdivide,abs(y_ch(:,iy,:)),sigma2(:,iy,:)))` for laplace?